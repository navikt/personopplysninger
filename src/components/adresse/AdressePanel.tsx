import * as React from "react";
import { Undertittel } from "nav-frontend-typografi";
import { FormattedMessage } from "react-intl";

interface Props {
  tittel: string;
  children: JSX.Element | JSX.Element[];
}

const AdressePanel = (props: Props) => (
  <div className="addresse__box">
    <div className="underseksjon__header">
      <Undertittel>
        <FormattedMessage id={props.tittel} />
      </Undertittel>
    </div>
    {props.children}
  </div>
);
export default AdressePanel;
