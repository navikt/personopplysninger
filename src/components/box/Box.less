.nav-veileder__frame {
  background: transparent !important;
}

.box {
  &__wrapper {
    padding-top: 1rem;
  }

  &__container {
    @media @desktop {
      padding: 0 3rem 1rem 3rem;
    }
  }

  &__title-container {
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
  }

  &__ikon {
    width: 72.5% !important;
  }

  &__title-i-icon {
    cursor: pointer;
  }

  &__icon-container {
    @media @desktop {
      padding: 0 1.5rem 0 0;
    }
  }

  &__content-container {
    width: 100%;

    @media @tablet {
      padding: 1.75rem 0;
    }
  }

  &__header {
    display: flex;
    flex-flow: column;
  }

  &__header > * {
    margin-bottom: 0.4rem;
  }

  &__linje-smal {
    border: 1px solid black;
    width: 15%;
  }

  &__linje-bred {
    border: 0.5px solid black;
    width: 100%;
    opacity: 0.5;
    margin: 1rem auto;
  }

  &__content {
    ul.list-column-2 {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;
      list-style-type: none;
      padding-left: 0;
      margin: 0;

      > li {
        width: 50%;
        flex-grow: 1;
        box-sizing: border-box;
        padding: 0.5rem 2rem 0.5rem 0rem;

        .title {
          display: block;
          white-space: nowrap;
          text-overflow: ellipsis;
          overflow: hidden;
        }

        @media @mobile {
          width: 100%;
        }
      }
    }
  }
}
