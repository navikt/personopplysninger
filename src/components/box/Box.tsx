import React from "react";
import { Systemtittel } from "nav-frontend-typografi";
import { FormattedMessage, injectIntl, InjectedIntlProps } from "react-intl";
import Veilederpanel from "nav-frontend-veilederpanel";
import Modal from "nav-frontend-modal";
import Infotekst from "../infotekst/Infotekst";

interface Props {
  id: string;
  tittel: string;
  beskrivelse?: string;
  icon?: string;
  children: JSX.Element | JSX.Element[];
}

Modal.setAppElement("#app");
const Box = (props: Props & InjectedIntlProps) => {
  const { tittel, beskrivelse, icon, children, id } = props;

  return (
    <div className="box__wrapper">
      <Veilederpanel
        svg={<img src={icon} className="box__ikon" alt="Veileder" />}
        type={"plakat"}
        kompakt
      >
        <div className="box__container">
          <div className="box__header" id={id}>
            <div className="box__title-container">
              {tittel && (
                <Systemtittel>
                  <FormattedMessage id={tittel} />
                </Systemtittel>
              )}
              {beskrivelse && <Infotekst beskrivelse={beskrivelse} />}
            </div>
          </div>
          <div className="box__content">{children}</div>
        </div>
      </Veilederpanel>
    </div>
  );
};

export default injectIntl(Box);
