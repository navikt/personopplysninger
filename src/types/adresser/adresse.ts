export type Adresse =
    // TPS
    | PostadresseIFrittFormat
    | UtenlandskAdresseIFrittFormat
    // PDL
    | Vegadresse
    | Postboksadresse
    | UtenlandskAdresse
    | Matrikkeladresse
    | Ukjentbosted;

// TPS
export interface PostadresseIFrittFormat {
    type: 'POSTADRESSE_I_FRITT_FORMAT';
    adresselinje1?: string;
    adresselinje2?: string;
    adresselinje3?: string;
    postnummer?: string;
    poststed?: string;
    coAdressenavn?: string;
}

export interface UtenlandskAdresseIFrittFormat {
    type: 'UTENLANDSK_ADRESSE_I_FRITT_FORMAT';
    adresselinje1?: string;
    adresselinje2?: string;
    adresselinje3?: string;
    postkode?: string;
    byEllerStedsnavn?: string;
    landkode?: String;
    land?: string;
}

// PDL
export interface Vegadresse {
    type: 'VEGADRESSE';
    husnummer?: string;
    husbokstav?: string;
    bruksenhetsnummer?: string;
    adressenavn?: string;
    kommunenummer?: string;
    kommune?: string;
    tilleggsnavn?: string;
    postnummer?: string;
    poststed?: string;
    coAdressenavn?: string;
}

export interface Postboksadresse {
    type: 'POSTBOKSADRESSE';
    postbokseier?: string;
    postboks: string;
    postnummer?: string;
    poststed?: string;
    coAdressenavn?: string;
}

export interface UtenlandskAdresse {
    type: 'UTENLANDSK_ADRESSE';
    adressenavnNummer?: string;
    bygningEtasjeLeilighet?: string;
    postboksNummerNavn?: string;
    postkode?: string;
    bySted?: string;
    regionDistriktOmraade?: string;
    landkode: string;
    land: string;
    coAdressenavn?: string;
}

export interface Matrikkeladresse {
    type: 'MATRIKKELADRESSE';
    bruksenhetsnummer?: string;
    tilleggsnavn?: string;
    postnummer?: string;
    poststed?: string;
    kommunenummer?: string;
    kommune?: string;
    coAdressenavn?: string;
}

export interface Ukjentbosted {
    type: 'UKJENTBOSTED';
    bostedskommune?: string;
    coAdressenavn?: string;
}
