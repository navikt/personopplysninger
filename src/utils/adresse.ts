export const slaSammenAdresse = (adresse1: string, adresse2: string, adresse3: string) => `${adresse1 || ''}${adresse2 || ''}${adresse3 || ''}`;

export default slaSammenAdresse;
