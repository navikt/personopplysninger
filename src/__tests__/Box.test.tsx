import * as React from "react";
import ReactTestRenderer from "react-test-renderer";
import wrapIntl from "./__utils__/IntlTestHelper";
import Box from "../components/box/Box";
import adresse from "../assets/img/Adresse.svg";

test("render Header with content", () => {
  const component = ReactTestRenderer.create(
    wrapIntl(
      <Box
        id="adresse"
        tittel="adresse.tittel"
        beskrivelse="adresse.beskrivelse"
        icon={adresse}
      >
        <h3>Header</h3>
        <div>content</div>
      </Box>
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});
