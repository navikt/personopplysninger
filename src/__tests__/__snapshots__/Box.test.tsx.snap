// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header with content 1`] = `
<div
  className="box__container seksjon"
>
  <div
    className="box__icon-container icon__container"
    id="adresse"
  >
    <div
      className="icon__circle"
      style={
        Object {
          "backgroundColor": undefined,
          "backgroundImage": "url(function SvgAdresser(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgAdresser.name})
  );
})",
        }
      }
    />
  </div>
  <div
    className="box__content-container"
  >
    <div
      className="box__header"
      id="adresse"
    >
      <h2
        className="typo-systemtittel"
      >
        <span>
          Adresser
        </span>
      </h2>
      <div
        className="box__ingress"
      >
        <p
          className="typo-ingress"
        >
          <span>
            Her finner du bostedsadresse og eventuelt din midlertidige adresse.
          </span>
        </p>
      </div>
    </div>
    <div
      className="box__content"
    >
      <h3>
        Header
      </h3>
      <div>
        content
      </div>
    </div>
  </div>
</div>
`;
